make_my_trip/lib/features/booking/presentation/cubit/payment_integeration_cubit.dart

```
@@ -0,0 +1,150 @@
+
+class PaymentCubit extends Cubit<BaseState> {
+  PaymentCubit(
+      this._razorpay,
+      this.paymentUseCase,
+      this.bookingUseCase,
+      this.roomBookPostUsecase,
+      this.showApplicableCouponsUsecase,
+      this.checkCouponUsecase)
+      : super(StateInitial());
+
+  final Razorpay _razorpay;
+  final PaymentUseCase paymentUseCase;
+  final BookingUseCase bookingUseCase;
+  final RoomBookPostUsecase roomBookPostUsecase;
+  final ShowApplicableCouponsUsecase showApplicableCouponsUsecase;
+  final CheckCouponUsecase checkCouponUsecase;
+
+  init() {
+    _razorpay.on(Razorpay.EVENT_PAYMENT_SUCCESS, _handlePaymentSuccess);
+    _razorpay.on(Razorpay.EVENT_PAYMENT_ERROR, _handlePaymentError);
+    _razorpay.on(Razorpay.EVENT_EXTERNAL_WALLET, _handleExternalWallet);
+  }
+
+  void _handlePaymentSuccess(PaymentSuccessResponse response) {
+    emit(StateShowSearching());
+    Fluttertoast.showToast(
+        msg: "SUCCESS: ${response.paymentId}", timeInSecForIosWeb: 4);
+  }
+
+  void _handlePaymentError(PaymentFailureResponse response) {
+    emit(StateNoData());
+    Fluttertoast.showToast(
+        msg: "ERROR: ${response.code} - ${response.message}",
+        timeInSecForIosWeb: 4);
+  }
+
+  void _handleExternalWallet(ExternalWalletResponse response) {
+    Fluttertoast.showToast(
+        msg: "EXTERNAL_WALLET: ${response.walletName}", timeInSecForIosWeb: 4);
+  }
+
+  void openCheckout(double amount, String orderId, String? name, String? email,
+      String? number) async {
+    var options = {
+      'key': 'rzp_test_LqRkyAt6cQpVui',
+      'amount': amount,
+      'name': name,
+      'order_id': orderId,
+      'description': 'Payment',
+      'prefill': {
+        'contact': number ?? "8888888888",
+        'email': email ?? "testuser@gmail.com"
+      },
+      'external': {
+        'wallets': ['paytm']
+      }
+    };
+
+    try {
+      _razorpay.open(options);
+    } catch (e) {}
```

why exception is not handled here?